### PR TITLE
add CUAV smart battery support

### PR DESCRIPTION
**Pull request by mxiaogit**
Add a UAVCAN V0 protocol for CUAV smart battery.

### CODE REVIEW

**Comment by dagar** [comment]
This looks good, but do you want to bring it in with UAVCAN modifications to actually use the message?

**Comment by mxiaogit** [comment]
@dagar Is there any problem?

File: msg/battery_status.msg [hunk 1]
```diff
@@ -43,3 +43,12 @@ uint8 warning						# current battery warning
 
 
 uint8 MAX_INSTANCES = 4
+
+float32 average_power               # The average power of the current discharge
+float32 available_energy            # The predicted charge or energy remaining in the battery
+float32 remaining_capacity          # The compensated battery capacity remaining
+float32 full_charge_capacity        # The compensated capacity of the battery when fully charged
```
Comment (dagar) [inline msg/battery_status.msg:50]: There seems to be some overlap here with existing fields like `capacity`.

Comment (mxiaogit) [reply msg/battery_status.msg:50]: What are the specific it can be considered to overlap? remaining_capacity, full_charge_capacity and capacity?@dagar

**Comment by dagar** [comment]
Can you resolve the merge conflicts?

**Comment by mxiaogit** [comment]
> Can you resolve the merge conflicts?

**Comment by mxiaogit** [comment]
> > Can you resolve the merge conflicts?
@dagar The merge conflict should have been resolved.

**Comment by dagar** [comment]
I'm happy to merge this as is right now, but I'd still like to see if there's a way we could transition to an existing or new standard battery message. https://github.com/UAVCAN/public_regulated_data_types/blob/legacy-v0/uavcan/equipment/power/1092.BatteryInfo.uavcan

**Review by dagar — APPROVED** [review]

**Pull request merged**